### PR TITLE
[#94] addnation (connects #94)

### PR DESCRIPTION
**Pull request by Pr0chin**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "karma-jasmine": "~1.1.0",
     "karma-jasmine-html-reporter": "^0.2.2",
     "node-sass": "^4.5.3",
+    "popper.js": "^1.12.5",
```
Comment (lmmrssa) [inline package.json]: this should not be in this issue

File: src/app/home/home-router.module.ts [hunk 2]
```diff
@@ -5,12 +5,15 @@ import { DashboardComponent } from '../dashboard/dashboard.component';
 import { UsersComponent } from '../users/users.component';
 import { HomeComponent } from './home.component';
 import { CoursesComponent } from '../courses/courses.component';
+import { NationComponent } from '../nation/nation.component';
+
```
Comment (lmmrssa) [inline src/app/home/home-router.module.ts]: looks like extra line

File: src/app/home/home.module.ts [hunk 3]
```diff
@@ -11,7 +11,8 @@ import { HomeRouterModule } from './home-router.module';
 import { CoursesComponent } from '../courses/courses.component';
 import { FormErrorMessagesComponent } from '../form-error-messages/form-error-messages.component';
 
-import { CourseValidatorService } from '../validators/course-validator.service';
+import { CourseValidatorService } from '../validators/course-validator.service';
```
Comment (lmmrssa) [inline src/app/home/home.module.ts]: why is this line showing as change

File: src/app/nation/nation.component.html [hunk 4]
```diff
@@ -0,0 +1,53 @@
+
+  <h1>Add Nation </h1>
+  <form class="form-horizontal"(ngSubmit)="onSubmit(nation.value)" #nation="ngForm">
+    <div class="form-group">
+      <label for="admin_name">Admin Name</label>
+      <input type="text" style="width:50%;" class="form-control" id="admin_name" required minlength="5" maxlength="30" [(ngModel)]="nation.admin_name" name="admin_name" #admin_name="ngModel">
```
Comment (lmmrssa) [inline src/app/nation/nation.component.html]: better not to put inline style

File: src/app/nation/nation.component.html [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      <div *ngIf="admin_name.errors && (admin_name.dirty || admin_name.touched)" class="alert alert-danger" style="width:50%;">
+        <div [hidden]="!admin_name.errors.required">
```
Comment (lmmrssa) [inline src/app/nation/nation.component.html]: It probably can be changed better way. @LKhadka had something that adds error message later based on validation

Comment (LKhadka) [reply src/app/nation/nation.component.html]: This needs to be changed into a reactive form then you can use the validators [here](https://github.com/ole-vi/planet/blob/master/src/app/validators/custom-validators.ts). @Pr0chin can see an example [here](https://github.com/ole-vi/planet/blob/master/src/app/courses/courses.component.ts)

File: src/app/nation/nation.component.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+@Component({
+  selector: 'app-nation',
+  templateUrl: './nation.component.html',
+  styleUrls: ['./nation.component.scss']
+})
+export class NationComponent implements OnInit {
+message = '';
+  constructor(
+  	private couchService: CouchService,
```
Comment (paulbert) [inline src/app/nation/nation.component.ts]: These should be indented to 4, not 3 spaces

File: src/app/nation/nation.component.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  	private router: Router
+
+  ) { }
+  ngOnInit() {
+  }
+  onSubmit(nation) {
+    if (nation.nation_name !== '' && nation.nationurl !== '' && nation.type !=="") {
+    console.log(nation.nation_name, nation.nationurl, nation.type);
```
Comment (paulbert) [inline src/app/nation/nation.component.ts]: Should be indented within the if block

File: src/app/nation/nation.component.html (hunk 5 above)
Comment (Pr0chin) [reply src/app/nation/nation.component.html]: @LKhadka :+1:

File: package.json (hunk 1 above)
Comment (Pr0chin) [reply package.json]: ok :+1:

File: src/app/nation/nation.component.ts (hunk 6 above)
Comment (Pr0chin) [reply src/app/nation/nation.component.ts]: @paulbert  :+1:

File: package.json [hunk 8]
```diff
@@ -7,12 +7,12 @@
     "start": "LNG=${LNG:-en};ng serve --aot --i18n-file=src/i18n/messages.$LNG.xlf --locale=$LNG --i18n-format=xlf",
     "v-serve": "vagrant ssh -c 'cd /vagrant;ng serve'",
     "build": "ng build",
-    "v-build":"vagrant ssh -c 'cd /vagrant;ng build'",
+    "v-build": "vagrant ssh -c 'cd /vagrant;ng build'",
```
Comment (paulbert) [inline package.json]: Changes in this file are unrelated to the issue, so please remove them.

File: src/app/nation/nation.component.html [hunk 9]
```diff
@@ -0,0 +1,53 @@
+
+  <h1>Add Nation </h1>
+  <form class="form-horizontal"(ngSubmit)="onSubmit(nation.value)" #nation="ngForm">
+    <div class="form-group">
+      <label for="admin_name">Admin Name</label>
+      <input type="text" class="form-control" id="admin_name" required minlength="5" maxlength="30" [(ngModel)]="nation.admin_name" name="admin_name" #admin_name="ngModel">
+      <div *ngIf="admin_name.errors && (admin_name.dirty || admin_name.touched)" class="alert alert-danger" id = "danger">
```
Comment (paulbert) [inline src/app/nation/nation.component.html]: The `id` referenced here is not unique, which is invalid HTML.  Please use a `class` for this.

Also, the space around the equals should be removed.

File: src/app/nation/nation.component.html [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        </div>
+        <div [hidden]="!admin_name.errors.maxlength">
+          Admin name cannot be more than 30 characters long.
+        </div>
+      </div>
+    </div>
+     <div class="form-group">
+      <label for="nation_name">Name</label>
+      <input type="text" class="form-control" id="nation_name" required minlength="5" maxlength="30" [(ngModel)]="nation.nation_name" name="nation_name" #nation_name="ngModel">
+      <div *ngIf="nation_name.errors && (nation_name.dirty || nation_name.touched)" class="alert alert-danger"  id = "danger">
```
Comment (paulbert) [inline src/app/nation/nation.component.html]: Fix `id` same as above

File: src/app/nation/nation.component.html [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        </div>
+        <div [hidden]="!nation_name.errors.maxlength">
+           Name cannot be more than 30 characters long.
+        </div>
+      </div>
+    </div>
+    <div class="form-group">
+      <label for="nation_name">Nationurl</label>
+      <input type="text" class="form-control" id="nationurl" required minlength="5" maxlength="30" [(ngModel)]="nation.nationurl" name="nationurl" #nationurl="ngModel">
+      <div *ngIf="nationurl.errors && (nationurl.dirty || nationurl.touched)" class="alert alert-danger" id = "danger">
```
Comment (paulbert) [inline src/app/nation/nation.component.html]: Fix `id` same as above.

File: src/app/nation/nation.component.html [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+        <div [hidden]="!nationurl.errors.required">
+           NationUrl is is required!
+        </div>
+      </div>
+    </div>
+     <div class="form-group">
+      <label for="nation_name">Type</label>
+      <input type="text" class="form-control" id="type" required minlength="5" maxlength="30" [(ngModel)]="nation.type" name="type" #type="ngModel">
+      <div *ngIf="type.errors && (type.dirty || type.touched)" class="alert alert-danger" id = "danger">
```
Comment (paulbert) [inline src/app/nation/nation.component.html]: Fix `id` same as above.

File: src/app/nation/nation.component.scss [hunk 13]
```diff
@@ -0,0 +1,15 @@
+#admin_name {
```
Comment (paulbert) [inline src/app/nation/nation.component.scss]: This is very repetitive, and more fitting as a class in the main styles.scss since this is reusable.

File: src/app/nation/nation.component.ts [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  onSubmit(nation) {
+    if(nation.nation_name !== "" && nation.nationurl !== "" && nation.type !=="") {
+      this.couchService.post('nations', {'adminname':nation.admin_name, 'nationname': nation.nation_name,'nationurl':nation.nationurl, 'type':nation.type}, )
+        .then((data) => {
+        alert('Nation has been sucessfully created');
+        this.router.navigate(['']);
+      }, (error) => this.message = 'Error');
+    }else{
```
Comment (paulbert) [inline src/app/nation/nation.component.ts]: Please add spaces here to change it to `} else {`.

Before pushing code please make sure it runs through the linter either with the git hook or you can run it manually on your own.
